python/gui/auto_generated/qgsquerybuilder.sip.in

```diff
@@ -41,14 +41,33 @@ vector layer properties dialog
     QString sql();
     void setSql( const QString &sqlStatement );
 
+%If (HAVE_QSCI_SIP)
+
+    QgsCodeEditor *codeEditorWidget() const;
+%Docstring
+Returns the code editor widget for the SQL.
+
+.. versionadded:: 3.18
+%End
+%End
+%If (!HAVE_QSCI_SIP)
+
+    QWidget *codeEditorWidget() const;
+%Docstring
+Returns the code editor widget for the SQL.
+
+.. versionadded:: 3.18
+%End
+%End
+
   public slots:
     virtual void accept();
 
     virtual void reject();
 
     void clear();
 
-    void test();
+    virtual void test();
 %Docstring
 Test the constructed sql statement to see if the vector layer data provider likes it.
 The number of rows that would be returned is displayed in a message box.
```

Comment:
The docstring should reflect something like "the default implementation" ...

Reply:
done